coriolis/osmorphing/netpreserver/interfaces.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2025 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+import os
+
+from oslo_log import log as logging
+
+from coriolis.osmorphing.netpreserver import base
+
+LOG = logging.getLogger(__name__)
+
+
+class InterfacesNetPreserver(base.BaseNetPreserver):
+
+    def __init__(self, osmorphing_tool):
+            super(InterfacesNetPreserver, self).__init__(osmorphing_tool)
+            self.ifaces_file = "etc/network/interfaces"
+
+    def check_net_preserver(self):
+        if self.osmorphing_tool._test_path(self.ifaces_file):
+            if self.osmorphing_tool._list_dir(
+                    os.path.dirname(self.ifaces_file)):
+                return True
+        return False
+
+    def parse_network(self):
+        paths = [self.ifaces_file]
+
+        def _parse_iface_file(interface_file):
+            nonlocal paths
+            curr_ip_address = None
```

Comment:
Please also initialize `curr_iface` here. 
```suggestion
            curr_ip_address = None
            curr_iface = None
```